src/lib/Notification/Notification.php

```diff
@@ -0,0 +1,76 @@
+<?php
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\HybridPlatformUi\Notification;
+
+use eZ\Publish\API\Repository\Values\ValueObject;
+
+/**
+ * Notification object for messages to users, to be stored in a FlashBag.
+ * Can be cast to a string for sending as a Response but most use cases will just access properties of it.
+ *
+ * @property-read string $type
+ * @property-read string $message
+ * @property-read int $timeout
+ * @property-read bool $copyable
+ * @property-read string $details
+ */
+class Notification extends ValueObject
+{
+    const DEFAULT_TIMEOUT = 10;
+
+    const ERROR_TIMEOUT = 0;
+
+    const TYPE_SUCCESS = 'positive';
+
+    const TYPE_ERROR = 'error';
+
+    /**
+     * Message type.
+     *
+     * @var string
+     */
+    protected $type;
+
+    /**
+     * Message.
+     *
+     * @var string
+     */
+    protected $message;
+
+    /**
+     * Message timeout.
+     *
+     * @var int
+     */
+    protected $timeout;
+
+    /**
+     * Are details Copyable.
+     *
+     * @var bool
+     */
+    protected $copyable;
+
+    /**
+     * Extended details for Developers.
+     *
+     * @var string
+     */
+    protected $details;
+
+    public function __toString()
```

Comment:
I'm not sure we should keep that.

The rendering of a notification to a string can very well be done, using `sprintf` as well, from the `NotificationResponse`.